src/OrchardCore/OrchardCore/Modules/Extensions/RunningShellTableExtensions.cs

```diff
@@ -20,7 +20,7 @@ public static ShellSettings Match(this IRunningShellTable table, HttpContext htt
             // The same way .Scheme contains the protocol that the user set and not what a proxy
             // could be using (see X-Forwarded-Proto).
 
-            return table.Match(httpRequest.Host, httpRequest.Path, true);
+            return table.Match(httpRequest.Host, httpRequest.PathBase + httpRequest.Path, true);
```

Comment:
This extension method is intended to be called before the tenant router midlleware that executes the tenant pipeline.

When called in the tenant container middleware, the tenant prefix is not yet moved to the `PathBase`, it is still in `Path`.

https://github.com/OrchardCMS/OrchardCore/blob/e0e4dd79bfcfcecbcd4c6e52d08fea7f334e6baa/src/OrchardCore/OrchardCore/Modules/ModularTenantContainerMiddleware.cs#L31-L34

This is in the tenant router middleware, before executing the tenant pipeline, that it is moved to the `PathBase`.

https://github.com/OrchardCMS/OrchardCore/blob/e0e4dd79bfcfcecbcd4c6e52d08fea7f334e6baa/src/OrchardCore/OrchardCore/Modules/ModularTenantRouterMiddleware.cs#L46-L55

Reply:
I see, thanks for the explanation. Without this, you won't be able to use the extension method later though (this would've helped with https://github.com/Lombiq/Orchard-Azure-Application-Insights/issues/19). And it's not an issue if there's no `PathBase`, the concatenation works with an empty value too.

Reply:
> Without this, you won't be able to use the extension method later

Yes, exactly.

> And it's not an issue if there's no PathBase, the concatenation works with an empty value too.

But PathBase may be not empty, e.g. including a virtual folder.

Reply:
Regarding your other issue, normally injecting `ShellSettings` to retrieve its name should work, but need to be done in the context of a shell container, so may depend on where you are injecting `ITelemetryInitializer`.

Reply:
I didn't consider having a virtual folder. I don't know how to solve that `PathBase` may or may not contain what we need because I don't see how we can distinguish just based on the content of the `HttpContext`. However, if there is `IRunningShellTable.Match(HttpContext)` I'd expect it to work regardless of where we're in the request pipeline. Thus if it can't work generally then I'd make it be non-public, so it's not accessible for anybody and thus open to confusion.

I've solved the other issue with a workaround and it's not a huge problem anyway.

Reply:
Okay cool, yes maybe better to make this method internal.

So it's like something is resolved in a given tenant container but executed in the pipeline of another tenant => right prefix in pathBase, wrong name from the injected shellSettings. If so, just for info we have an httpContext feature that holds the following data including the shell from which you can retrieve the settings name.

https://github.com/OrchardCMS/OrchardCore/blob/e0e4dd79bfcfcecbcd4c6e52d08fea7f334e6baa/src/OrchardCore/OrchardCore/Modules/ModularTenantContainerMiddleware.cs#L52-L58

Reply:
Made it `internal` then. Thanks for the tip!